exonum-node/src/events/mod.rs

```diff
@@ -57,21 +67,39 @@ impl<T: Send + 'static> SyncSender<T> {
         use tokio::runtime::Handle;
 
         if let Ok(handle) = Handle::try_current() {
-            let mut sender = self.0.clone();
+            let mut sender = self.inner.clone();
+            let message_kind = self.message_kind;
             handle.spawn(async move {
-                sender.send(message).await.log_error();
+                if sender.send(message).await.is_err() {
+                    log::warn!(
+                        "Cannot send a {}; processing has shut down. {}",
+                        message_kind,
+                        Self::ERROR_MSG_SUFFIX
+                    );
+                }
             });
-        } else {
-            block_on(self.0.send(message)).log_error();
+        } else if block_on(self.inner.send(message)).is_err() {
```

Comment:
Does it make sense to unpack an error message from `block_on` invoke and to add to the log message ? Or it doesn't have a cognitive load ?